File: src/main/java/com/vandendaelen/nicephore/client/KeyMappings.java
package com.vandendaelen.nicephore.client;

import com.mojang.blaze3d.platform.InputConstants;
import com.vandendaelen.nicephore.Nicephore;
import net.minecraft.client.KeyMapping;
import net.minecraftforge.client.settings.KeyConflictContext;
import net.minecraftforge.client.settings.KeyModifier;
import org.lwjgl.glfw.GLFW;

public enum KeyMappings {
    COPY_KEY("copy", KeyConflictContext.IN_GAME, KeyModifier.CONTROL, GLFW.GLFW_KEY_C),
    GUI_SCREENSHOT_KEY("screenshots.gui", KeyConflictContext.IN_GAME, KeyModifier.CONTROL, GLFW.GLFW_KEY_S),
    GUI_GALLERY_KEY("gallery.gui", KeyConflictContext.IN_GAME, KeyModifier.CONTROL, GLFW.GLFW_KEY_G);

    private final KeyMapping key;

    KeyMappings(String localizationKey, KeyConflictContext context, KeyModifier modifier, int key) {
        this.key = new KeyMapping(Nicephore.MODID + ".keybinds." + localizationKey, context, modifier, getKey(key), Nicephore.MOD_NAME);
    }

    public boolean consumeClick() {
        return getKey().consumeClick();
    }

    private static InputConstants.Key getKey(final int key) {
        return InputConstants.Type.KEYSYM.getOrCreate(key);
    }

    public KeyMapping getKey() {
        return key;
    }
}

File: src/main/kotlin/com/vandendaelen/nicephore/client/KeyMappings.kt
package com.vandendaelen.nicephore.client

import com.mojang.blaze3d.platform.InputConstants
import com.vandendaelen.nicephore.Nicephore
import net.minecraft.client.KeyMapping
import net.minecraft.resources.Identifier
import net.neoforged.neoforge.client.settings.KeyConflictContext
import net.neoforged.neoforge.client.settings.KeyModifier
import org.lwjgl.glfw.GLFW

val NICEPHORE_CATEGORY = KeyMapping.Category(Identifier.fromNamespaceAndPath(Nicephore.MODID, "key_mapping"))

enum class KeyMappings(val key: KeyMapping) {
    COPY_KEY(
        KeyMapping(
            "${Nicephore.MODID}.keybinds.copy",
            KeyConflictContext.IN_GAME,
            KeyModifier.CONTROL,
            InputConstants.Type.KEYSYM.getOrCreate(GLFW.GLFW_KEY_C),
            NICEPHORE_CATEGORY
        )
    ),
    GUI_SCREENSHOT_KEY(
        KeyMapping(
            "${Nicephore.MODID}.keybinds.screenshots.gui",
            KeyConflictContext.IN_GAME,
            KeyModifier.CONTROL,
            InputConstants.Type.KEYSYM.getOrCreate(GLFW.GLFW_KEY_S),
            NICEPHORE_CATEGORY
        )
    ),
    GUI_GALLERY_KEY(
        KeyMapping(
            "${Nicephore.MODID}.keybinds.gallery.gui",
            KeyConflictContext.IN_GAME,
            KeyModifier.CONTROL,
            InputConstants.Type.KEYSYM.getOrCreate(GLFW.GLFW_KEY_G),
            NICEPHORE_CATEGORY
        )
    );

    fun consumeClick(): Boolean = key.consumeClick()
}
